tuner/tuner/libtuner.py

```diff
@@ -825,6 +830,69 @@ def generate_candidates(
     return candidates
 
 
+def generate_candidate_specs(
+    args: argparse.Namespace,
+    path_config: PathConfig,
+    candidate_trackers: list[CandidateTracker],
+) -> list[int]:
+    """Generate candidate transform dialect specs for tuning. Returns the list of candidate indexes"""
+    logging.debug("generate_candidate_specs()")
+
+    path_config.specs_dir.mkdir(parents=True, exist_ok=True)
+    tune_logger = logging.getLogger("tune")
+
+    # Generate transform dialect specs.
+    try:
+        with open(args.input_file, "r") as f:
+            mlir_text = f.read()
+        with ir.Context() as ctx:
+            tuner_context = TunerContext(ctx, tune_logger)
+            mlir_module = dispatch_parser.parse_mlir(mlir_text, tuner_context)
+            logging.debug("Captured messages from candidate_gen.py:")
+            config_specs: list[ir.Module] = candidate_gen.generate_configs_and_td_specs(
+                input_module=mlir_module,
+                tuner_context=tuner_context,
+                limit=args.num_candidates,
+                num_subgroups=args.num_subgroups,
+                # lhs_dims=args.lhs_dims,
+                # rhs_dims=args.rhs_dims,
+                # tile_dims=args.tile_dims,
```

Comment:
Oops, forgot to remove this. I'll remove it when addressing the first round of comments.